lib/ansible/modules/cloud/azure/azure_rm_subnet.py

```diff
@@ -257,6 +276,21 @@ def exec_module(self, **kwargs):
                     changed = True
                     results['route_table']['id'] = self.route_table
                     self.log("CHANGED: subnet {0} route_table to {1}".format(self.name, route_table['name']))
+
+                if self.service_endpoints:
+                    oldd = {}
+                    for item in self.service_endpoints:
+                        name = item['service']
+                        oldd[name] = {'service': name, 'locations': item['locations'].sort()}
+                    newd = {}
```

Comment:
what if service_endpoints not in results?

Reply:
… then change will be detected, as the newd dictionary will be empty

Reply:
oh, sorry.... needs one fix, thanks for catchin that!